File: reactors-core/src/main/scala/io/reactors/Observer.scala
package io.reactors






/** An object that can act upon an event or be signalled that
 *  there will be no more events.
 *
 *  @tparam T        type of events the observer responds to
 */
trait Observer[@spec(Int, Long, Double) T] {

  /** Called by an event stream when an event `value` is produced.
   * 
   *  @param value   the event passed to the observer
   */
  def react(value: T): Unit

  /** Called by an event stream when an exception is produced.
   *
   *  @param t       the exception passed to the observer
   */
  def except(t: Throwable): Unit

  /** Called by an event stream when there will be no further updates.
   */
  def unreact(): Unit

}

File: reactors-core/src/main/java/io/reactors/japi/Observer.java
package io.reactors.japi;



import java.util.function.Consumer;



public abstract class Observer<T> {
  private io.reactors.Observer<T> self;

  Observer(io.reactors.Observer<T> self) {
    this.self = self;
  }

  public void react(T value, Object hint) {
    self.react(value, hint);
  }

  public void except(Throwable t) {
    self.except(t);
  }

  public void unreact() {
    self.unreact();
  }
}
